tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -759,8 +803,44 @@ Status StorageManager::array_create(
       array_uri.join_path(constants::array_metadata_folder_name);
   RETURN_NOT_OK(vfs_->create_dir(array_metadata_uri));
 
+  std::string encryption_key_from_cfg;
+  std::string encryption_type_from_cfg;
+  EncryptionKey encryption_key_cfg;
+  EncryptionType encryption_type_cfg;
```

Comment:
These are not used outside of the if-statement on L513. Move them inside of that block.